### DISCUSSION TITLE
Hierarchical feature structure

### DISCUSSION THREAD

**Discussion by aleczratiu** [Unleash/unleash#4899] [General]
Hello,

I have a question regarding the implementation of a hierarchical feature structure, starting from a main feature.

For instance, in the context of constructing a building, I would like to enable the release of specific components once certain levels of apartments are completed, as follows:

```
unleash.tower = true
unleash.tower.level1 = true
unleash.tower.level2 = false

```

I have a few objectives with this approach:

1. Gradually deploy code into production to avoid the need for local feature maintenance.
2. Have the ability to manage sub-levels of the main feature in case of issues in the production environment.
3. Ensure that disabling the unleash.tower feature also deactivates its sub-levels.

Has anyone experimented with a similar approach before? Could anyone provide suggestions on how to implement or configure this in a feature management system like Unleash, or if it's feasible at all?

I found a way by having from the main feature a constraint and the underneath features to belong on that constrain if it is true or false will turn the level1 or level2 boolean based on the constraint from the main feature. Here I'm not sure if is right to do like this or are there other solutions.

**Comment by andreas-unleash** [comment]
Hi @aleczratiu , thanks for reaching out.  We are actually [building something that would solve your use case exactly.](https://github.com/Unleash/unleash/issues/2255)  Interesting workaround!

**Reply by aleczratiu** [reply]
I see, thank you for your time @andreas-unleash  it's bassically exactly what I need now I could work with my workaround for now or by having internally in team a rule for defining the unleash keys like `mainfeature.subfeature` and parsing by (.) can be traversed and checked one by one if one flag is disabled. 

I noticed the link is related to ["nextjs](https://github.com/Unleash/unleash/issues/2255) is merged in unleash nextjs client?

**Reply by andreas-unleash** [reply]
My pleasure! We are working on updating all the SDKs with the new functionality. Stay tuned :)
